include/nbl/video/IGPUDescriptorSet.h

```diff
@@ -112,49 +118,39 @@ class NBL_API IGPUDescriptorSet : public asset::IDescriptorSet<const IGPUDescrip
             return samplers + localOffset;
         }
 
-        inline uint32_t getDescriptorStorageOffset(const asset::E_DESCRIPTOR_TYPE type) const { return m_descriptorStorageOffsets.data[type]; }
-        inline uint32_t getMutableSamplerStorageOffset() const { return m_descriptorStorageOffsets.data[asset::EDT_COUNT]; }
-
-	protected:
-		virtual ~IGPUDescriptorSet()
-		{
-            for (auto i = 0u; i < asset::EDT_COUNT; ++i)
-            {
-                // There is no descriptor of such type in the set.
-                if (m_descriptorStorageOffsets.data[i] == ~0u)
-                    continue;
+        inline core::smart_refctd_ptr<asset::IDescriptor>* getAllDescriptors(const asset::IDescriptor::E_TYPE type) const
+        {
+            auto* baseAddress = m_pool->getDescriptorStorage(type);
+            if (baseAddress == nullptr)
+                return nullptr;
 
-                const auto type = static_cast<asset::E_DESCRIPTOR_TYPE>(i);
-                std::destroy_n(getDescriptorStorage(type) + m_descriptorStorageOffsets.data[i], m_layout->getTotalDescriptorCount(type));
-            }
+            const auto offset = m_descriptorStorageOffsets.data[static_cast<uint32_t>(type)];
+            if (offset == ~0u)
+                return nullptr;
 
-            const auto mutableSamplerCount = m_layout->getTotalMutableSamplerCount();
-            if (mutableSamplerCount > 0)
-                std::destroy_n(getMutableSamplerStorage() + getMutableSamplerStorageOffset(), mutableSamplerCount);
-		}
+            return baseAddress + offset;
+        }
 
-	private:
-        struct SDescriptorOffsets
+        inline core::smart_refctd_ptr<IGPUSampler>* getAllMutableSamplers() const
         {
-            SDescriptorOffsets()
-            {
-                // The default constructor should initiailze all the offsets to an invalid value (~0u) because ~IGPUDescriptorSet relies on it to
-                // know which descriptors are present in the set and hence should be destroyed.
-                std::fill_n(data, asset::EDT_COUNT + 1, ~0u);
-            }
+            auto* baseAddress = m_pool->getMutableSamplerStorage();
+            if (baseAddress == nullptr)
+                return nullptr;
+
+            const auto poolOffset = m_descriptorStorageOffsets.data[static_cast<uint32_t>(asset::IDescriptor::E_TYPE::ET_COUNT)];
+            if (poolOffset == ~0u)
```

Comment:
seems like you should be using `getDescriptorStorageOffset` for self documenting code :)

Reply:
I think using `getMutableSamplerStorageOffset` would be better. Resolved?